src/Domain/LeanCode.DomainModels/Model/Id.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Diagnostics;
+using LeanCode.DomainModels.Serialization;
+using LeanCode.IdentityProvider;
+
+namespace LeanCode.DomainModels.Model
+{
+    [DebuggerDisplay("{Value}")]
+    [TypedIdConverter]
+    public readonly struct Id<TEntity> : IEquatable<Id<TEntity>>, IComparable<Id<TEntity>>
+        where TEntity : class, IIdentifiable<Id<TEntity>>
+    {
+        public Guid Value { get; }
+
+        public Id(Guid value)
+        {
+            Value = value;
+        }
+
+        public static Id<TEntity> New() => new Id<TEntity>(Identity.NewId());
+        public static Id<TEntity> From(Guid id) => new Id<TEntity>(id);
+        public static Id<TEntity>? From(Guid? id) => id is Guid v ? new Id<TEntity>(v) : (Id<TEntity>?)null;
+
+        public bool Equals(Id<TEntity> other) => Value.Equals(other.Value);
+        public int CompareTo(Id<TEntity> other) => Value.CompareTo(other.Value);
+        public override bool Equals(object? obj) => obj is Id<TEntity> id && Value.Equals(id.Value);
+        public override int GetHashCode() => HashCode.Combine(Value);
+        public override string? ToString() => Value.ToString();
+
+        public static bool operator ==(Id<TEntity> left, Id<TEntity> right) => left.Equals(right);
+        public static bool operator !=(Id<TEntity> left, Id<TEntity> right) => !left.Equals(right);
+        public static implicit operator Guid(Id<TEntity> id) => id.Value;
+        public static explicit operator Id<TEntity>(Guid id) => new Id<TEntity>(id);
+    }
+
+    [DebuggerDisplay("{Value}")]
+    [TypedIdConverter]
+    public readonly struct IId<TEntity> : IEquatable<IId<TEntity>>, IComparable<IId<TEntity>>
+        where TEntity : class, IIdentifiable<IId<TEntity>>
+    {
+        public static readonly IId<TEntity> EmptyValue = IId<TEntity>.From(0);
```

Comment:
I'm not sure if we need that at all, since it's very easy to obtain such `EmptyValue` with `default(IId<TEntity>)` (and sometimes you will be able to omit the type too, if it can be inferred automagically). Or, if there _is_ need for such static field, then maybe add one to `Guid`–based Id type too?

Reply:
This part was copy pasted and I overlooked it :stuck_out_tongue: 
I guess we can add it to `Id` as well to mimic they way Guid behaves, but your point with `default` is valid.